### PR TITLE
Load testing

### PR DESCRIPTION
**Pull request by mariubg**

### CODE REVIEW

File: tests/load_tests/locustfile.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    @task
+    def ask_question(self):
+        test_username, test_password = self.credentials()
+        response = self.client.post(
+            "/auth/token", data={"username": test_username, "password": test_password}
+        )
+        token = response.json()["access_token"]
+
+        response = self.client.post(
+            "/chatbot/",
```
Comment (petr7555) [inline tests/load_tests/locustfile.py:42]: trailing slash

File: tests/load_tests/locustfile.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        response = self.client.post(
+            "/auth/token", data={"username": test_username, "password": test_password}
+        )
+        token = response.json()["access_token"]
+        response = self.client.get("/conversation", headers={"Authorization": f"Bearer {token}"})
+
+    @task
+    def ask_question(self):
+        test_username, test_password = self.credentials()
+        response = self.client.post(
```
Comment (petr7555) [inline tests/load_tests/locustfile.py:36]: Won't calling the auth endpoint distort the test? We need to know the time of question-answering.

**Comment by petr7555** [comment]
Can this be setup as a test? I.e. you set the values (latency) the service must meet, some scenarios such as continuous requests, spikes, etc. – and at the end there will be a pass/fail result?

File: pyproject.toml [hunk 3]
```diff
@@ -28,6 +28,7 @@ unstructured = {extras = ["pdf"], version = "^0.10.15"}
 pymupdf = "^1.23.3"
 chromadb = "^0.4.13"
 azure-cosmos = "^4.5.1"
+locust = "^2.17.0"
```
Comment (petr7555) [inline pyproject.toml:31]: Should be a dev dependency.

File: README.md [hunk 4]
```diff
@@ -47,6 +47,11 @@ AUTH_PASSWORD=any_password
 
 - `poetry run pytest`
 
+### Load tests
```
Comment (petr7555) [inline README.md:50]: Which server do these tests use?

File: tests/load_tests/locustfile.py (hunk 2 above)
Comment (mariubg) [reply tests/load_tests/locustfile.py:36]: Locust logs the time of each request, not each test, so the test should not be distorted. So we will get two separate times. I've made a page in notion to record the results from one test, as an example of how the results look (I only ran the ask_question test): https://www.notion.so/LoadTesting-e32ea69c73a3491dafd9c4384899b579

File: README.md (hunk 4 above)
Comment (mariubg) [reply README.md:50]: The server you want. Running locust opens a web ui where you can specify stuff like duration, number of users, and the server you want to test. So you can test both deployment and local servers without having to alter the code

**Comment by petr7555** [comment]
Superseded by https://github.com/petr7555/ai-document-search-backend/pull/23.

**Pull request closed without merging**